crates/compiler/builtins/roc/Hash.roc

```diff
@@ -73,12 +75,38 @@ Hasher has
     ## accumulated hash state.
     complete : a -> U64 | a has Hasher
 
+    ## Resets the internal state of a hasher
+    ## The hasher should still have all the same parameters and seeds.
+    reset : a -> a | a has Hasher
```

Comment:
What do you think of naming this `newFrom` or similar, or at least changing the doc to say something like

```suggestion
    ## Creates a new hasher with the internal state reset.
    ## The new hasher should have all the same parameters and seeds as the original.
    reset : a -> a | a has Hasher
```

? I don't want to bike shed too much here (I know I'm instigating, sorry 😅 ) but when I read the current comment I immediately jumped to think this did some in-place stuff, despite the type annotation - maybe that's just my bias though. I'll defer to your judgement.

Reply:
removed function as we discussed.